css/src/atomics/border.scss

```diff
@@ -92,12 +92,51 @@
 
 // Colors
 
+@mixin border-gradient($bgColor, $colorStart, $colorEnd) {
+	border-color: transparent !important;
+	background: linear-gradient($bgColor 0 0) padding-box,
+		linear-gradient(90deg, $colorStart 0%, $colorEnd 100%) border-box !important;
+}
+
 @each $name, $color-set in $colors {
 	$base: nth($color-set, $color-index-base);
+	$dark: nth($color-set, $color-index-dark);
+	$background: nth($color-set, $color-index-background);
+	$invert: nth($color-set, $color-index-invert);
 
 	.border-color-#{$name} {
 		border-color: $base !important;
 	}
+
+	// handle hard-coded background colors like background-color-alternate
+
+	@each $gradientName, $gradientSet in $gradients {
```

Comment:
Let's chat about this in person, I don't think we want to produce this much output for a single border.